tests/flex/test_executablecontainer.py

```diff
@@ -100,9 +100,9 @@ def test_job_files(self):
             self.assertTrue(file in dir(job.files))
         output_file_path = os.path.abspath(os.path.join(__file__, "..", "test_executablecontainer", "job_output_files_hdf5", "job_output_files", "error.out"))
         if os.name != "nt":
-            self.assertEqual(job.files.error_out, output_file_path)
+            self.assertEqual(str(job.files.error_out), output_file_path)
```

Comment:
This makes it very confusing for the user. For example when we take a look at the NFDI example we want to be able to access the path of the files directly: https://github.com/BAMresearch/NFDI4IngScientificWorkflowRequirements/blob/main/exemplary_workflow/pyiron/workflow.py#L34

Reply:
We can fix this by supporting the `File` object as input for restart_files and convert it to a string there.

Reply:
`input_file_lst=[gmsh.files.square_msh.path]` would be an option. I don't think it would be very confusing, just extra typing.  I've tried to look through the places where `restart_file_lst/dict` are used and we could force a conversion there somewhere, but I'd be afraid I'll miss places.  I'll revert the commits that remove it sub class if you don't see an easy path in between as well.

Reply:
I provided a suggestion in https://github.com/pyiron/pyiron_base/pull/1397